### PR TITLE
feat: clin_sig spell

### PR DESCRIPTION
**Pull request by BenOrl**
New spell to visualize the clinical significance

### CODE REVIEW

**Comment by fxwiegand** [comment]
Just as a general info: There is no need to open up a new PR if there is a new change. Simply push another commit to the branch of the currently opened PR and the one already open will update.

File: sign/clin_sig/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+name: clin_sig
+type: "column consistent of strings, separated by ',' "
```
Comment (johanneskoester) [inline sign/clin_sig/meta.yaml:2]:
```suggestion
type: "column consisting of strings, separated by ',' "
```

File: sign/clin_sig/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+description: |
+  This spell visualizes the clinical significance, given in clinvar significance terms
+link-to-url: https://www.ncbi.nlm.nih.gov/clinvar/
```
Comment (johanneskoester) [inline sign/clin_sig/meta.yaml:5]: Is this used in the docs? Maybe rather annotate the mention of clinvar in the description above with a markdown link.

File: sign/clin_sig/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+example: 
+  code: |
+    render-table:
+      columns: 
+        Significance Column:
```
Comment (johanneskoester) [inline sign/clin_sig/meta.yaml:10]: The column name is not optimal. Maybe rather "clinical significance"?

**Review by johanneskoester — CHANGES_REQUESTED** [review]

**Comment by FelixMoelder** [comment]
To get the tests passing some changes are required. Here is a summary as I do not have write access and github suggestions would get a bit out of hand.

- `test_*.yaml` and `data*.tsv` must be placed in a subdirectory called `tests`
- `test*.yaml`:
  -  files need an underscore e.g `test_*.yaml`
  - `url` and `path` need to be relative to the yaml-file's location 
  - the column name should match the one set in the `data*.tsv`
- `data*.tsv`: choose a more generic column name as this is just an example e.g. `some clinical significance column` (optional) 
- `meta.yaml`:
  - `type` needs to be `column` as it is not a description field
  - check for correct indentation and spelling of `test-data` 
  - adjust the path after moving the yaml-file into a subdirectory 
- `spell.yaml`: the function must not have a name to be called

File: sign/clin_sig/tests/test_1.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+      separator: "\t"
+
+views:
+  test:
+    dataset: test
+    render-table:
+      columns: 
+        some clinical significance column:
+          spell:
+            url: sign/clin_sig/spell.yaml
```
Comment (fxwiegand) [inline sign/clin_sig/tests/test_1.yaml:15]:
```suggestion
            url: ../spell.yaml
```

File: sign/clin_sig/tests/test_2.yaml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+      separator: "\t"
+
+views:
+  test:
+    dataset: test
+    render-table:
+      columns: 
+        some clinical significance column:
+          spell:
+            url: sign/clin_sig/spell.yaml
```
Comment (fxwiegand) [inline sign/clin_sig/tests/test_2.yaml:15]:
```suggestion
            url: ../spell.yaml
```

File: sign/clin_sig/tests/test_3.yaml [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+      separator: "\t"
+
+views:
+  test:
+    dataset: test
+    render-table:
+      columns: 
+        some clinical significance column:
+          spell:
+            url: sign/clin_sig/spell.yaml
```
Comment (fxwiegand) [inline sign/clin_sig/tests/test_3.yaml:15]:
```suggestion
            url: ../spell.yaml
```

File: sign/clin_sig/tests/test_1.yaml [hunk 7]
```diff
@@ -0,0 +1,15 @@
+name: Test
+
+datasets:
+    test:
+      path: sign/clin_sig/tests/data1.tsv
```
Comment (fxwiegand) [inline sign/clin_sig/tests/test_1.yaml:5]:
```suggestion
      path: data1.tsv
```

File: sign/clin_sig/tests/test_2.yaml [hunk 8]
```diff
@@ -0,0 +1,15 @@
+name: Test2
+
+datasets:
+    test:
+      path: sign/clin_sig/tests/data2.tsv
```
Comment (fxwiegand) [inline sign/clin_sig/tests/test_2.yaml:5]:
```suggestion
      path: data2.tsv
```

File: sign/clin_sig/tests/test_3.yaml [hunk 9]
```diff
@@ -0,0 +1,15 @@
+name: Test3
+
+datasets:
+    test:
+      path: sign/clin_sig/tests/data3.tsv
```
Comment (fxwiegand) [inline sign/clin_sig/tests/test_3.yaml:5]:
```suggestion
      path: data3.tsv
```

**Review by fxwiegand — CHANGES_REQUESTED** [review]
The paths to the files containing the test data also need be be relative - sorry I should've noticed that in the last review 😅

File: sign/clin_sig/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+  This spell visualizes the clinical significance, given in clinvar significance terms (https://www.ncbi.nlm.nih.gov/clinvar/)
+  The values should be given in a column consisting of strings and separated by ','
+example: 
+  code: |
+    render-table:
+      columns: 
+        some clinical significance column:
+          spell:
+            url: <url>
+  test-data: sign/clin_sig/tests/data1.tsv
```
Comment (FelixMoelder) [inline sign/clin_sig/meta.yaml:13]:
```suggestion
  test-data: tests/data1.tsv
```

**Comment by FelixMoelder** [comment]
To me this looks good now.

File: default.profraw
Comment (FelixMoelder) [inline default.profraw:1]: This should also be removed

**Comment by FelixMoelder** [comment]
@johanneskoester should the directory more contextual like `bio/clin_sig` or `annotation/clin_sig`?

**Comment by johanneskoester** [comment]
> @johanneskoester should the directory more contextual like `bio/clin_sig`, ~`vep/clin_sig`~ or `annotation/clin_sig`?

`bio/clin-sig` I would say.

**Review by FelixMoelder — APPROVED** [review]

**Comment by FelixMoelder** [comment]
Looks got to me.

**Review by fxwiegand — APPROVED** [review]

**Pull request merged**